topology/graph/indexer.go

```diff
@@ -196,26 +198,19 @@ func (m *MetadataIndexer) Get(values ...interface{}) ([]*Node, []interface{}) {
 
 // NewMetadataIndexer returns a new metadata graph indexer for the nodes
 // matching the graph filter `m`, indexing the metadata with `indexes`
-func NewMetadataIndexer(g *Graph, m GraphElementMatcher, indexes ...string) (indexer *MetadataIndexer) {
+func NewMetadataIndexer(g *Graph, listenerHandler GraphListenerHandler, m GraphElementMatcher, indexes ...string) (indexer *MetadataIndexer) {
```

Comment:
why do we need the extra (separate) `listenerHandler` what is the use case for it not using `g *Graph` ?

Reply:
k8s.ResourceCache (previously kubeCache) is a GraphListenerHandler. This way, we do not receive all the events on the graph but only the events of the ResourceCache (when a Service, a Pod, ... is created)

Reply:
ok